src/platform/packages/shared/kbn-esql-utils/src/utils/query_parsing_helpers.ts

```diff
@@ -276,3 +283,60 @@ export const fixESQLQueryWithVariables = (
 
   return queryString;
 };
+
+export const getCategorizeColumns = (esql: string): string[] => {
+  const { root } = parse(esql);
+  const statsCommand = root.commands.find(({ name }) => name === 'stats');
+  if (!statsCommand) {
+    return [];
+  }
+  const options: ESQLCommandOption[] = [];
+  const columns: string[] = [];
+
+  walk(statsCommand, {
+    visitCommandOption: (node) => options.push(node),
+  });
+
+  const statsByOptions = options.find(({ name }) => name === 'by');
```

Comment:
```suggestion
  const columns: string[] = [];

  const statsByOption: ESQLCommandOption = Walker.find(statsCommand, (node) => node.type === 'option' && node.name === 'by');
```

Reply:
I see... yeah that's annoying.

Reply:
If we got rid of the AST-node-can-be-an-array it would clean up so much stuff.

Reply:
Yeah I agree, we need to cast in many places using our apis. Improving this will make the DX much better